rpc/packetizer.go

```diff
@@ -67,6 +69,12 @@ func (p *packetHandler) NextFrame() (rpcMessage, error) {
 		return nil, NewPacketizerError("invalid frame size: %d", len(bytes))
 	}
 
+	// Log the bytes as the last thing, just in case the logger
+	// mutates those bytes (which it shouldn't).
+	defer func() {
+		p.log.FrameRead(bytes)
```

Comment:
I would rather every single call to `FrameRead` also has some kind of `ShouldFrameTrace` check in the API, so it's blatantly obvious that every single potential instance of logging a frame's bytes is guarded.

To be clear, that check should _also_ happen in the logger. But additionally having it here would be less scary.